validator/src/command.rs

```diff
@@ -0,0 +1,226 @@
+        }
+    }
+}
+
+impl Display for Values {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+pub fn parse_config() -> Option<Config> {
+    let args = Cli::parse();
+    match args.command {
+        Commands::Run(cmd) => {
+            let config_file = config_file();
+            let config_data = fs::read_to_string(config_file).expect(
+                "Failed to read config file, make sure u run init command \
+                 before you try to run.",
+            );
+            let default_config: Config = toml::from_str(&config_data).unwrap();
+            let keypair = if cmd.keypair_path.is_some() {
+                let keypair = read_keypair_file(&cmd.keypair_path.unwrap())
+                    .expect("Unable to read keypair file");
+                keypair.to_bytes().to_vec()
+            } else {
+                default_config.keypair
+            };
+            let config = Config {
+                rpc_url: cmd.rpc_url.unwrap_or(default_config.rpc_url),
+                ws_url: cmd.ws_url.unwrap_or(default_config.ws_url),
+                program_id: cmd.program_id.unwrap_or(default_config.program_id),
+                genesis_hash: cmd
+                    .genesis_hash
+                    .unwrap_or(default_config.genesis_hash),
+                keypair,
+                log_level: cmd
+                    .log_level
+                    .unwrap_or(LevelFilter::Info)
+                    .to_string(),
+            };
+            setup_logging(&config.log_level);
+            Some(config)
+        }
+        Commands::Init(cmd) => {
+            let config_file = config_file();
+            let file = fs::metadata(config_file.clone());
+            setup_logging(
+                &cmd.log_level.unwrap_or(LevelFilter::Info).to_string(),
+            );
+            if file.is_ok() {
+                let value = prompt(
+                    "Do you really want to overwrite. Enter yes or no.",
+                    Some(Values::No),
+                );
+                if matches!(value, Values::No) {
+                    log::info!("Skipped overwriting the config file");
+                    return None;
+                }
+                log::info!("Overwriting config file");
+            }
+            let keypair = read_keypair_file(&cmd.keypair_path)
+                .expect("Unable to read keypair file");
+            let keypair = keypair.to_bytes().to_vec();
+            let config = Config {
+                rpc_url: cmd.rpc_url,
+                ws_url: cmd.ws_url,
+                program_id: cmd.program_id,
+                genesis_hash: cmd.genesis_hash,
+                keypair,
+                log_level: cmd
+                    .log_level
+                    .unwrap_or(LevelFilter::Info)
+                    .to_string(),
+            };
+            let toml_in_string = toml::to_string(&config).unwrap();
```

Comment:
to_string_pretty? Though to be honest I have no idea what’s the difference between the two.

Reply:
well for the `toml` i dont think it makes any difference because it seems to be formatted with just `to_string`.

Reply:
You can do that with a derive.  I don’t really have strong opinions.  For a moment I wondered whether we should be using base64 to encode the keys but then it looks like Solana code generally encodes keys as array of bytes.

Reply:
Did u meant this comment for the above conversation? If so, the `Keypair` from solana doesnt implement `Serde` so we may have to write the traits.

Reply:
Yes, sorry.  So yeah, do whatever you feel is best.  Using Keypair might have an advantage that we don’t have to convert it into something else later and that it gets validated when deserialising.  We could use `#[serde(serialize_with = "...")]` and `#[serde(deserialize_with = "...")]` to implement the serialisation.  At this point it probably ends up being more code but it would be better typed code.

Reply:
Well sounds good to me. Will write a wrapper type.